lib/sles4sap/aws_cli.pm

```diff
@@ -0,0 +1,495 @@
+  aws_create_vm
+  aws_get_vm_id
+  aws_wait_instance_status_ok
+  aws_get_ip_address
+);
+
+
+=head2 aws_import_key_pair
+
+    aws_import_key_pair($ssh_key, $pub_key_path);
+
+Import an SSH public key pair into AWS EC2 for instance authentication
+
+=over
+
+=item B<$ssh_key> - name to assign to the imported key pair in AWS
+
+=item B<$pub_key_path> - filesystem path to the public key file
+
+=back
+=cut
+
+sub aws_import_key_pair {
+    my ($ssh_key, $pub_key_path) = @_;
+    assert_script_run("aws ec2 import-key-pair --key-name '$ssh_key' --public-key-material fileb://$pub_key_path");
+}
+
+=head2 aws_create_vpc
+
+    my $vpc_id = aws_create_vpc($region, $cidr, $job_id);
+
+Create a new AWS VPC with a specified CIDR block and tag it with the OpenQA job ID
+
+=over
+
+=item B<$region> - AWS region where to create the VPC
+
+=item B<$cidr> - CIDR block for the VPC (e.g., '10.0.0.0/16')
+
+=item B<$job_id> - OpenQA job identifier for tagging
+
+=back
+
+Returns the VPC ID
+=cut
+
+sub aws_create_vpc {
+    my ($region, $cidr, $job_id) = @_;
+    my $vpc_id = script_output("aws ec2 create-vpc --region $region --cidr-block $cidr --query 'Vpc.VpcId' --output text");
+    die('VPC creation failed: VPC ID is empty') unless $vpc_id;
+    assert_script_run("aws ec2 create-tags --resources $vpc_id --tags Key=OpenQAJobVpc,Value=$job_id --region $region");
+    return $vpc_id;
+}
+
+=head2 aws_get_vpc_id
+
+    my $vpc_id = aws_get_vpc_id($region, $job_id);
+
+Retrieve the VPC ID associated with a specific OpenQA job
+
+=over
+
+=item B<$region> - AWS region where the VPC is located
+
+=item B<$job_id> - OpenQA job identifier used to tag the VPC
+
+=back
+
+Returns the VPC ID
+=cut
+
+sub aws_get_vpc_id {
+    my ($region, $job_id) = @_;
+    return script_output(
+        "aws ec2 describe-vpcs --filters 'Name=tag:OpenQAJobVpc,Values=$job_id'" .
+          " --region $region --query 'Vpcs[0].VpcId' --output text");
+}
+
+=head2 aws_create_security_group
+
+    my $sg_id = aws_create_security_group($region, $group_name, $description, $vpc_id);
+
+Create an AWS security group within a VPC and tag it with the OpenQA job ID
+
+=over
+
+=item B<$region> - AWS region where to create the security group
+
+=item B<$group_name> - name for the security group
+
+=item B<$description> - description of the security group purpose
+
+=item B<$vpc_id> - ID of the VPC where the security group will be created
+
+=item B<$job_id> - OpenQA job identifier used to tag the VPC
+
+=back
+
+Returns the security group ID
+=cut
+
+sub aws_create_security_group {
+    my ($region, $group_name, $description, $vpc_id, $job_id) = @_;
+    my $sg_id = script_output(
+        "aws ec2 create-security-group --region $region --group-name $group_name " .
+          "--description '$description' --vpc-id $vpc_id --query 'GroupId' --output text"
+    );
+    assert_script_run("aws ec2 create-tags --resources $sg_id --tags Key=OpenQAJobSg,Value=$job_id --region $region");
```

Comment:
Please add a validation for `sg_id` before using it to compose next command `aws ec2 create-tags`